app/gui/src/project-view/components/GraphEditor/GraphNode.vue

```diff
@@ -434,6 +438,20 @@ const nodeClass = computed(() => {
   }
 })
 
+const nodeProgress = computed(() => evaluationProgress(expressionInfo.value) ?? 100)
+const { getProgressStyles, watchProgress } = useProgressBackground(nodeProgress, {
+  counterId: () => expressionInfo.value?.evaluationId,
+})
+const { progressAnimating, backgroundProgressEvents } = watchProgress()
+
+// The progress bar animation and the color change animation are incompatible. When progress is not
+// partial or animating, we stop applying progress styles and fall back to the background color set
+// by the class, so that its CSS transition will be used if the type changes. (If the type changes
+// during progress animation, we'll just change colors abruptly).
+function backgroundProgressStyles() {
```

Comment:
Could this be a computed value? Otherwise the whole logic executes on every render and causes `style` property to be unnecessarily diffed (because a distinct object is returned each time).